### PR TITLE
docs: improve getting started experience

### PR DESCRIPTION
**Pull request by bmorelli25**
## Summary

This PR brings us closer to the Observability documentation structure described in **Outline for Observability** (internal Google Doc). There are three main changes:

* Improved getting started experience:
  * Makes the getting started guide a single topic on a single page (instead of four!)
  * Adds the tabbed widget for displaying multi-platform commands
  * Adds basic installation instructions for each Agent
* Rename `Release highlights` to `What's new` and move to the top of the TOC. Why?...
  * This structure is easier to maintain than multiple files.
  * Google analytics show us that "What's new" info gets more hits than "release highlights"
  * We want users to have a consistent user experience across docs. This aligns our approach with Beats, ES, and Kib.
* Simplifies the top-level TOC by grouping like topics together

## Screenshots

<details><summary>Quick start</summary><be>

screencapture-localhost-8000-guide-install-and-run-html-2020-09-22-08_13_06

</details>

<details><summary>TOC</summary><be>

Screen Shot 2020-09-22 at 8 12 54 AM

</details>

## Related issues

Closes https://github.com/elastic/apm/issues/277.

## Follow-ups

- [ ] A PR must be opened to change the location of the cross-doc includes and the `apm-release-notes` link in the [Installation and Upgrade guide](https://www.elastic.co/guide/en/elastic-stack/master/apm-highlights.html). After the change, `apm-release-notes.asciidoc` can be deleted.
- [ ] `whats-new.asciidoc` needs to be updated in `7.x` and `7.9` (expand details for text)

<details><summary>Details</summary><be>

```asciidoc
// 8.0
// [[whats-new]]
// == What's new in {minor-version}

// coming[8.0.0]

// Here are the highlights of what's new and improved in {minor-version}.
// For detailed information about this release,
// check out the {apm-server-ref-v}/release-notes.html[APM Server Release Notes] and
// {kibana-ref}/release-notes.html[Kibana Release Notes].

// 7.10
// [[whats-new]]
// == What's new in {minor-version}

// coming[7.10.0]

// Here are the highlights of what's new and improved in {minor-version}.
// For detailed information about this release,
// check out the {apm-server-ref-v}/release-notes.html[APM Server Release Notes] and
// {kibana-ref}/release-notes.html[Kibana Release Notes].

// Other versions: {apm-ref-all}/7.9/whats-new.html[7.9] | {apm-ref-all}/7.8/release-highlights-7.8.0.html[7.8] |
// {apm-ref-all}/7.7/release-highlights-7.7.0.html[7.7] | {apm-ref-all}/7.6/release-highlights-7.6.0.html[7.6] |
// {apm-ref-all}/7.5/release-highlights-7.5.0.html[7.5] | {apm-ref-all}/7.4/release-highlights-7.4.0.html[7.4] |
// {apm-ref-all}/7.3/release-highlights-7.3.0.html[7.3] | {apm-ref-all}/7.2/release-highlights-7.2.0.html[7.2] |
// {apm-ref-all}/7.1/release-highlights-7.1.0.html[7.1] | {apm-ref-all}/7.0/release-highlights-7.0.0.html[7.0]
```
</details>

### CODE REVIEW

**Review by simitt — COMMENTED** [review]
The screenshot looks great!

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/tab-widgets/install-agents.asciidoc [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+* {apm-java-ref-v}/configuration.html[Advanced configuration]
+// end::java[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::net[]
+*Download the APM agent*
+
+Add the the agent package(s) from https://www.nuget.org/packages?q=Elastic.apm[NuGet] to your .NET application.
```
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/tab-widgets/install-agents.asciidoc:97]: Parens make content harder to read. It's OK to use plural to mean one or more. If you mean only one, use singular.

File: docs/tab-widgets/install-agents.asciidoc [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+// tag::rum[]
+*Enable Real User Monitoring support in APM Server*
+
+APM Server disables RUM support by default.
+To enable it, set `apm-server.rum.enabled: true` in your APM Server configuration file.
+
+*Set up the Agent as a dependency*
+
+You can install the Agent as a dependency to your application with `npm install @elastic/apm-rum --save`.
```
Comment (dedemorton) [inline docs/tab-widgets/install-agents.asciidoc:385]: I've noticed a little inconsistency across these tabs wrt how you refer to the agent. Sometimes you say APM agent, sometimes agent (lowercase), and sometimes Agent uppercase. It's kind of minor but might be worth fixing in this PR.

File: docs/tab-widgets/install-agents.asciidoc [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+  // Set service version (required for source map feature)
+  serviceVersion: ''
+})
+----
+
+Framework integrations, like React or Angular, have custom dependencies.
+See {apm-rum-ref-v}/framework-integrations.html[framework integrations] for more information.
+
+*Set up the Agent with Script Tags*
```
Comment (dedemorton) [inline docs/tab-widgets/install-agents.asciidoc:408]: Does "Script Tags" need to be capitalized?

File: docs/tab-widgets/install-agents.asciidoc [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+
+Alternatively, you can use Script tags to set up and configure the Agent.
```
Comment (dedemorton) [inline docs/tab-widgets/install-agents.asciidoc:410]: same

File: docs/tab-widgets/install-agents.asciidoc (hunk 3 above)
Comment (dedemorton) [reply docs/tab-widgets/install-agents.asciidoc:408]: So it sounds like the user has two options: set up the agent as a dependency or use script tags? If possible, I'd make that more obvious so users know up front that there are different ways to do this.

File: docs/guide/install-and-run.asciidoc [hunk 5]
```diff
@@ -1,269 +1,104 @@
...
-Just follow the steps below.
-
-**Create a docker-compose.yml file**
+// tag::configure-agents[]
 
-The https://www.docker.elastic.co/[Elastic Docker registry] contains Docker images for all of the products
-in the {stack}.
-You can use Docker compose to easily get the default distributions of {es}, {kib},
-and APM Server up and running in Docker.
+Now that you're up and running with Elastic APM, you may want to adjust some configuration settings.
```
Comment (dedemorton) [inline docs/guide/install-and-run.asciidoc:61]: I wonder...before I get to this part, is there some way for me to confirm that I am indeed "up and running"?

File: docs/guide/features.asciidoc [hunk 6]
```diff
@@ -0,0 +1,18 @@
+[[apm-features]]
+== Elastic APM features
```
Comment (dedemorton) [inline docs/guide/features.asciidoc:2]: I like how these are grouped here!

**Review by dedemorton — COMMENTED** [review]
Big improvement! I think the tabs work nicely for providing a *lot* of related content in a single place.

File: docs/guide/install-and-run.asciidoc (hunk 5 above)
Comment (bmorelli25) [reply docs/guide/install-and-run.asciidoc:61]: Good point.

File: docs/tab-widgets/install-agents.asciidoc [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+# Use if APM Server requires a token
+secret_token: ''
+
+# Set custom APM Server URL (default: http://localhost:8200)
+server_url: 'http://localhost:8200'
+----
+
+Rack::
++
+For Rack or a compatible framework (e.g. Sinatra), include the middleware in your app and start the agent.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:324]: I agree. The Agent instructions were copied from Kibana and I never proofed them 😬

Comment (bmorelli25) [inline docs/tab-widgets/install-agents.asciidoc:324]:
```suggestion
For Rack or a compatible framework, like Sinatra, include the middleware in your app and start the agent.
```

File: docs/tab-widgets/install-agents.asciidoc [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+    "ElasticApm": {
+    "SecretToken": "",
+    "ServerUrls": "http://localhost:8200", //Set custom APM Server URL (default: http://localhost:8200)
+    "ServiceName" : "MyApp", //allowed characters: a-z, A-Z, 0-9, -, _, and space. Default is the entry assembly of the application
+  }
+}
+----
+
+In case you don’t pass an `IConfiguration` instance to the agent (e.g. in case of non ASP.NET Core applications)
+you can also configure the agent through environment variables.
```
Comment (bmorelli25) [inline docs/tab-widgets/install-agents.asciidoc:146]:
```suggestion
If you don’t pass an `IConfiguration` instance to the agent, for example, in a non-ASP.NET Core application,
you can configure the agent with environment variables.
```

File: docs/tab-widgets/install-agents.asciidoc [hunk 9]
```diff
@@ -0,0 +1,433 @@
...
+{
+    "ElasticApm": {
+    "SecretToken": "",
+    "ServerUrls": "http://localhost:8200", //Set custom APM Server URL (default: http://localhost:8200)
+    "ServiceName" : "MyApp", //allowed characters: a-z, A-Z, 0-9, -, _, and space. Default is the entry assembly of the application
+  }
+}
+----
+
+In case you don’t pass an `IConfiguration` instance to the agent (e.g. in case of non ASP.NET Core applications)
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:145]: It's a fine line. Let me look into it.

File: docs/tab-widgets/install-agents.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:384]: 👍

File: docs/tab-widgets/install-agents.asciidoc [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+
+  // Set service version (required for source map feature)
+  serviceVersion: ''
+})
+----
+
+Framework integrations, like React or Angular, have custom dependencies.
+See {apm-rum-ref-v}/framework-integrations.html[framework integrations] for more information.
+
+*Set up the Agent with Script Tags*
```
Comment (bmorelli25) [inline docs/tab-widgets/install-agents.asciidoc:407]:
```suggestion
*Set up the Agent with `<script>` tags*
```

File: docs/tab-widgets/install-agents.asciidoc [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+
+// ***************************************************
+// ***************************************************
+
+// tag::rum[]
+*Enable Real User Monitoring support in APM Server*
+
+APM Server disables RUM support by default.
+To enable it, set `apm-server.rum.enabled: true` in your APM Server configuration file.
+
```
Comment (bmorelli25) [inline docs/tab-widgets/install-agents.asciidoc:381]:
```suggestion

With RUM support enabled, you can now set up the RUM Agent.
There are two ways to do this: add the agent as a dependency,
or set it up with `<script>` tags.

```

File: docs/tab-widgets/install-agents.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:407]: 👍

File: docs/tab-widgets/install-agents.asciidoc (hunk 4 above)
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:409]:
```suggestion
Alternatively, you can use `<script>` tags to set up and configure the Agent.
```

**Review by bmorelli25 — COMMENTED** [review]
Thanks, DeDe! Definitely should have proofed those Agent instructions before copying them in 😓

File: docs/tab-widgets/install-agents.asciidoc [hunk 12]
```diff
@@ -0,0 +1,438 @@
+// tag::go[]
```
Comment (axw) [inline docs/tab-widgets/install-agents.asciidoc:1]: Given relative language popularity: would it be sensible (and possible) to retain alphabetical ordering, but make the Java agent the initially active tab?

**Review by axw — APPROVED** [review]
Love it :)

File: docs/tab-widgets/install-agents.asciidoc (hunk 12 above)
Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:1]: An interesting idea! Let me play around with it.

Comment (bmorelli25) [reply docs/tab-widgets/install-agents.asciidoc:1]: I like it. Done. 🙇

**Pull request merged**